### PR TITLE
Allergies18 march24

### PR DESCRIPTION
**Pull request by RashiTiwari21**

### CODE REVIEW

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	/**
+	 * Locators
+	 */
+	@FindBy(xpath = "//div[@class='content-container node node--slide node--full node--slide--full']")
+	WebElement bannerImg;
+	
+	@FindBy(css = "h1#content-main")
+	WebElement bannerHeading;
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: @RashiTiwari21 all locators should be private

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ComparePage.java [hunk 2]
```diff
@@ -197,7 +197,7 @@ public void verifyRelatedProducts(String heading, String product, String expecte
 			Action.explicitWait(buyNowBtn(product), 30);
 			buyNowBtn(product).click();
 			extentInfoLog("Clicked on : ", "buy now button");
-			signup.closeSignUpPopup();
+			Action.waitFor(2000);
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ComparePage.java]: Why is a static sleep placed here ?
@RashiTiwari21

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java [hunk 3]
```diff
@@ -30,25 +31,22 @@ public SignUpPage() {
 	 * Locators
 	 */
 	@FindBy(xpath = "//div[@class='sfmc-careclub-logo']")
-	private WebElement logo;
+	WebElement logo;
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java]: All private declarations are removed ?
@RashiTiwari21

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		extentInfoLog("Content Scrolled to : ", Action.isDisplayed(driver, getAllergyScroll(headingScroll)));
+	}
+
+	/**
+	 * Verifies if the related content card is displayed for the specified index.
+	 * 
+	 * @param index The index of the related content card to verify.
+	 */
+	public void verifyRelatedContent(String index) {
+		extentInfoLog("Content card is displayed : ", Action.isDisplayed(driver, getRelatedArticleCard(index)));
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: Where is the assertion @RashiTiwari21 ?

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	/**
+	 * Verifies if the related product is displayed.
+	 */
+	public void verifyRelatedProduct() {
+		extentInfoLog("Related Product is displayed : ", Action.isDisplayed(driver, relatedProduct));
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: Where is the assertion @RashiTiwari21 ?

**Review by goutamnaik007 — CHANGES_REQUESTED** [review]
**Not all the tests as expected are covered.**
Please work on below scenarios
- Cover redirections for references
- Verify related products 
- Verify contents of different "Common Causes of Allergies"
- Verify more about links under allergies
- Check redirections for related articles

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java (hunk 5 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: @goutamnaik007 removed this function and in the test file calling relatedProduct() from compare page.

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java (hunk 4 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: @goutamnaik007 removed this function as it contains only isDisplayed and added verifyArticles()

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java (hunk 3 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java]: All the locators added as private @goutamnaik007

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ComparePage.java (hunk 2 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ComparePage.java]: @RashiTiwari21 removed this static wait and added signup.closeSignUpPopup();

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java (hunk 1 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: @goutamnaik007  all locators are private

Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: @vaibhavNagvekarSJI Verified

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ComparePage.java (hunk 2 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ComparePage.java]: Verified @RashiTiwari21

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java (hunk 3 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java]: @vaibhavNagvekarSJI Verified

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java (hunk 5 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: Verifed @vaibhavNagvekarSJI

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java (hunk 4 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java]: Verified @vaibhavNagvekarSJI

**Pull request merged**